### ISSUE TITLE
TS3 Handshake Failed

### ISSUE THREAD

**Issue by xenvious** [Itokoyamato/TokoVOIP_TS3#187]
Hello. The last two days I've been experiencing something very odd with TokoVOIP. Each time I join a server, I receive the same error as shown below.

![Error](https://i.gyazo.com/3049e12e229e2c0ce6202e24004b7457.png)

I've tried each step listed in the troubleshooting tips, to no avail. So I started doing some logging to look into the issue further when I noticed something very odd.

In my Teamspeak Client Logs, I see the following:
![TS3](https://i.gyazo.com/e96d58724a334a16f943c77fa1730ffb.png)

No problem there. Looks like it's trying to get the handshake with my IP address.

On a fresh ws_server install, I enabled the logging feature in the config.js and see the following:
![FIVEM](https://i.gyazo.com/e531ddb3982b26eef81bed6b57e76506.png)

If we decode that Base64 string, we get `107.107.57.218`. This is an unknown IP that is completely unrelated to my server and/or client. I've tried to ping this IP, and there is simply nothing there.

So, it appears at this point, FiveM is getting & sending the wrong IP address with the handshake to the Teamspeak plugin, not allowing the handshake to succeed.

I've been working on this issue for the last two days, and it's getting increasingly frustrating being unable to join any TokoVOIP server. Nobody else on my server is experiencing this issue.

Any help or guidance is appreciated.

**Comment by Itokoyamato** [comment]
It's probably not the wrong ip, just that you have a dynamic IP that changes over time. #179 1.5.5 will support this.
Hopefully thats just the issue here

**Comment by xenvious** [comment]
Is there a chance of any temporary fixes until 1.5.5 comes out?

**Comment by Itokoyamato** [comment]
You'll have to research your router / isp to know if you can have a static IP. I can't help you on that side

**Comment by xenvious** [comment]
After the latest release of 1.5.5 and the dynamic IPs, It appears this issue is still occurring. It's almost like the IP is getting encoded/decoded improperly in Base64.

**Comment by Oblivii** [comment]
Looks like the developer doesn’t care to look into this issue.

**Comment by odinnmar** [comment]
ya'll want the soloution to this ?

**Comment by odinnmar** [comment]
whether you have it figured out or not.
what worked for me was downloading a VPN service and connecting through that and BOOM no problem ts3 tokovoip connects like there is no tomorow

I didnt know that this would actually work, so i downloaded a free vpn and am currently still using this one for free
https://protonvpn.com/download

FYI. I done reinstalled windows TWICE and rubbed my head in this issue for 2 days before this was solved

**Comment by DanyCryEn** [comment]
Have any fix for this?.... same issue with 1.5.6

**Comment by BeadyRaindrop** [comment]
> Have any fix for this?.... same issue with 1.5.6

Same...

**Comment by odinnmar** [comment]
> > Have any fix for this?.... same issue with 1.5.6
> 
> Same...

Did you try using a vpn service? I posted above

**Comment by BeadyRaindrop** [comment]
No, i didn't. Although this would probably fix the issue, im not planning on using a VPN everytime. + All free vpn's are slow 😜

**Comment by xenvious** [comment]
> No, i didn't. Although this would probably fix the issue, im not planning on using a VPN everytime. + All free vpn's are slow 😜

A VPN actually does not fix the issue. The issue lies with your ISP blocking the handshake all together. Using a VPN just adds an extra step, but the handshake still has to be completed through your ISP. I’ve found that this ultimately happens on untraditional internet connections that don’t handle P2P traffic well. (Hotspots, etc)

**Comment by odinnmar** [comment]
For me it was like something was blocked. So when i used the vpn it unblocked. And after that with or without vpn tokovoip always works
